helpers/test/test_lib_tasks_docker_release.py

```diff
@@ -0,0 +1,175 @@
+import logging
+import re
+import unittest.mock as umock
+
+import helpers.hsystem as hsystem
+import helpers.hunit_test as hunitest
+import helpers.lib_tasks_docker_release as hltadore
+
+_LOG = logging.getLogger(__name__)
+
+
+# #############################################################################
+# TestDockerBuildLocalImage1
+# #############################################################################
+
+
+class TestDockerBuildLocalImage1(hunitest.TestCase):
+
+    def setUp(self) -> None:
+        """
+        Set up test environment and initialize all necessary mocks.
+        """
+        # Mock system calls.
+        self.system_patcher = umock.patch("helpers.hsystem.system")
+        self.mock_system = self.system_patcher.start()
+        # Mock run.
+        self.run_patcher = umock.patch("helpers.lib_tasks_utils.run")
+        self.mock_run = self.run_patcher.start()
+        # Mock version validation.
+        self.version_patcher = umock.patch(
+            "helpers.lib_tasks_docker.dassert_is_subsequent_version"
+        )
+        self.mock_version = self.version_patcher.start()
+        # Mock docker login.
+        self.docker_login_patcher = umock.patch(
+            "helpers.lib_tasks_docker.docker_login"
+        )
+        self.mock_docker_login = self.docker_login_patcher.start()
+        self.user = hsystem.get_user_name()
+
+    def tearDown(self) -> None:
+        """
+        Clean up test environment by stopping all mocks after each test case.
+        """
+        self.system_patcher.stop()
+        self.run_patcher.stop()
+        self.version_patcher.stop()
+        self.docker_login_patcher.stop()
+
+    def test_docker_build_single_arch(self) -> None:
+        """
+        Test building a local Docker image with single architecture.
+
+        This test verifies that the correct sequence of commands is
+        generated for building a local Docker image with single
+        architecture.
+        """
+        # Prepare inputs.
+        mock_ctx = umock.MagicMock()
+        test_version = "1.0.0"
+        test_base_image = "test-registry.com/test-image"
+        test_multi_arch = ""
+        # Call tested function.
+        hltadore.docker_build_local_image.body(
+            ctx=mock_ctx,
```

Comment:
we use `param=...` only for optional params, so you shouldn't need them for a few of them